### DISCUSSION TITLE
CRASHES

### DISCUSSION THREAD

**Discussion by kochikame** [junzis/pyModeS#160] [Q&A]
Hi again
It crashes when messages arrives
(ValueError('too many values to unpack (expected 3)'), 'Traceback (most recent call last):\n  File "/home/oscar/miniconda3/lib/python3.11/site-packages/pyModeS/streamer/decode.py", line 278, in run\n    self.process_raw(\n  File "/home/oscar/miniconda3/lib/python3.11/site-packages/pyModeS/streamer/decode.py", line 158, in process_raw\n    ac["HPL"], ac["RCu"], ac["RCv"] = pms.adsb.nuc_p(msg)\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nValueError: too many values to unpack (expected 3)\n')

**Comment by xoolive** [comment]
Indeed that part of the code is not statically typed-checked, and the function returns a tuple of 4 elements.

You should be able to fix it as:
```python
_, ac["HPL"], ac["RCu"], ac["RCv"] = pms.adsb.nuc_p(msg)
```

I am not doing it right now as I have to focus on more urgent tasks and would prefer the full script to be typed-checked on next commit.

**Reply by kochikame** [reply]
great, now does not stall!
With this version can I listen to mode S only, not ADS-B?
I am commercial pilot and flight instructor.
As I said before I want to build a portable device to detect nearby planes when I flight the piper PA28, in Europe ADS-B is not implemented in light planes so they are only use mode S only
Hope I can develop it and share with all you guys.
